test/image/mocks/bar_text_relative.json

```diff
@@ -0,0 +1,40 @@
+{
+    "data":[
+        {
+            "width":[1,0.8,0.6,0.4],
+            "text":[1,2,3,4],
+	    "textposition":"outside",
+            "y":[1,2,3,4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":[0.4,0.6,0.8,1],
+            "text":["Three",2,"This bar is size one",0],
+	    "textposition":"auto",
+            "y":[3,2,1,0],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":1,
+            "text":[-1,-3,-2,-4],
+	    "textposition":"inside",
+            "y":[-1,-3,-2,-4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "text":[0,"very small size",-3,-2],
```

Comment:
is there a criterion by which we should rotate an outside label ('very small size') to vertical like you do with inside labels ('Three')?

Also, and this applies to all the inside labels, I'd have expected inside labels to be pushed next to the bar end, the same way as you have outside labels right on the other side of the bar end. It looks like [some people](http://stackoverflow.com/questions/6644997/showing-data-values-on-stacked-bar-chart-in-ggplot2) want the labels in the middle, so perhaps this should be an option (perhaps textalign: middle or end) but I think semantically it's a better default to draw people's eyes toward the bar end. Seems like that's [somewhat more common](https://www.google.com/search?q=bar+chart+with+text&espv=2&biw=1568&bih=842&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjDnuKj6pnQAhUO3GMKHf8HCIIQ_AUIBigB&dpr=1#tbm=isch&q=bar+chart+with+text+inside) as well.